### ISSUE TITLE
ESLint plugin + VS Code ESLint extension on Windows results in `npm ci` erroring due to nx-win32-x64-msvc.node in use

### ISSUE THREAD

**Issue by CreativeTechGuy** [nrwl/nx#20346]
### Current Behavior

When VS Code is open, the NX ESLint plugin is in use and the VS Code ESLint extension is active, run `npm ci`. This command will error and npm will exit resulting in a broken node_modules folder. (partially uninstalled)

### Expected Behavior

The use of the @nx/eslint-plugin in VS Code wouldn't result in npm failing to remove/reinstall packages.

### Steps to Reproduce

A combination of a few things must happen for this issue to show up, but this is a fairly reasonable setup so isn't unique to me in any way.

Preconditions:

* Be using VS Code on Windows
* Use the [ESLint Extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
* Have an ESLint config active which uses the [@nx/eslint-plugin](https://nx.dev/nx-api/eslint-plugin)

Problem: 

* When VS Code is open and the VS Code ESLint extension is active, run `npm ci`. This will error and npm will exit with a broken node_modules folder (partially uninstalled)

### Nx Report

```shell
Node   : 20.9.0
   OS     : win32-x64
   npm    : 10.1.0
 
   nx                 : 17.1.1
   @nx/js             : 17.1.1
   @nx/jest           : 17.1.1
   @nx/linter         : 17.1.1
   @nx/eslint         : 17.1.1
   @nx/workspace      : 17.1.1
   @nx/devkit         : 17.1.1
   @nx/esbuild        : 17.1.1
   @nx/eslint-plugin  : 17.1.1
   @nx/node           : 17.1.1
   @nx/playwright     : 17.1.1
   @nx/plugin         : 17.1.1
   @nrwl/tao          : 17.1.1
   @nx/vite           : 17.1.1
   @nx/web            : 17.1.1
   typescript         : 5.2.2
```

### Failure Logs

```shell
> npm ci
npm ERR! code EPERM
npm ERR! syscall unlink
npm ERR! path C:\Users\username\repo\node_modules\@nx\nx-win32-x64-msvc\nx.win32-x64-msvc.node
npm ERR! errno -4048
npm ERR! Error: EPERM: operation not permitted, unlink 'C:\Users\username\repo\node_modules\@nx\nx-win32-x64-msvc\nx.win32-x64-msvc.node'
npm ERR!  [Error: EPERM: operation not permitted, unlink 'C:\Users\username\repo\node_modules\@nx\nx-win32-x64-msvc\nx.win32-x64-msvc.node'] {
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'unlink',
npm ERR!   path: 'C:\\Users\\username\\repo\\node_modules\\@nx\\nx-win32-x64-msvc\\nx.win32-x64-msvc.node'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It's possible that the file was already in use (by a text editor or antivirus),
npm ERR! or that you lack permissions to access it.
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.
```
```


### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

This is the only ESLint plugin I've ever seen that has such an issue. It seems to be related to how native bindings are handled on Windows. It seems to not be an issue with WSL so it seems like it's Windows specific. It appears that the native bindings aren't strictly necessary and could be swapped out with non-native implementations. If the root issue cannot be fixed, it'd be great if there was at least an option to not use the native bindings and fall back to a pure-Node.js implementation.

**Comment by csutorasr** [comment]
Have you tried to run `nx reset` to stop every instance of `nx` before removing the files? Also it might be worthy to run it from a console while `npm ci` removes `node_modules` including every executable.

**Comment by CreativeTechGuy** [comment]
@csutorasr Hi! Thanks for following up. No that doesn't work. It doesn't change anything to run `nx reset` before `npm ci`. Same issue occurs.

**Comment by mklueh** [comment]
Same here, can't use `nx add` because of that. `nx reset` did not help

**Comment by Sandwell** [comment]
Same here

**Comment by ertl** [comment]
Happens to me with IntelliJ and PNPM.

I have tried the following:

- set `NX_DAEMON=false`
- deactivate NX Console Plugin
- calling nx directly via `pnpx `

But without success.

It looks like it occurs whenever I run the command  `pnpm install` of the generated `package.json`. 
I use the `@nx/webpack:webpack` executor.

**Comment by DanielSchnieders** [comment]
Same Problem here. I can not run `yarn install --force` succesfully because the files are blocked. 
I can not delete the files on windows when vscode is open. I have to close it.

We are not directly using `@nx` . We are only using Lerna, which nowadays brings some nx dependencies with it.

For example `@lerna/create@8.1.2` has `"@nx/devkit" ">=17.1.2 < 19"` as it's dependency.

**Comment by ertl** [comment]
Is there any progress on this issue or is there a workaround. Killing all NodeJS processes and manually cleaning up the directory leads to a very poor developer experience?

**Comment by MaxKless** [comment]
hey everyone! 
This was fixed with https://github.com/nrwl/nx/pull/22648 
We now load the .node files from the `.nx/cache` folder instead of `node_modules`. This means you'll be able to run `npm ci` without issue again.

You can try it out right away with one of the nx 19 beta versions.
Please reach out if you still run into any errors.

**Comment by DanielSchnieders** [comment]
> hey everyone! This was fixed with #22648 We now load the .node files from the `.nx/cache` folder instead of `node_modules`. This means you'll be able to run `npm ci` without issue again.
> 
> You can try it out right away with one of the nx 19 beta versions. Please reach out if you still run into any errors.

How can we try it out? We are not using NX directly. See 3 posts above.

**Comment by MaxKless** [comment]
ah sorry, I missed that. You will have to wait until lerna releases a new version based on nx 19

**Comment by CreativeTechGuy** [comment]
@MaxKless Now `nx reset` fails with the same error. I'm guessing because that command tries to delete the `.nx/cache` folder.

**Comment by MaxKless** [comment]
@CreativeTechGuy did you try this on the beta? And what is your cache directory set to?
The default should be `.nx/cache`, not `node_modules/.nx/cache`

**Comment by CreativeTechGuy** [comment]
@MaxKless Ahh yeah my cache was set to node_modules. I've removed it to let it be default.

And no I didn't try it on the beta. It looks like that change was actually released in [18.3.2](https://github.com/nrwl/nx/releases/tag/18.3.2) so I tried it with the 18.3.3 (which was the latest at the time of the comment above). That's when I noticed the above error.

After changing back to the default cache location, it still fails when running `nx reset`.

**Comment by CreativeTechGuy** [comment]
Hi @MaxKless, should I open a new issue for this? https://github.com/nrwl/nx/issues/20346#issuecomment-2081250902

**Comment by MaxKless** [comment]
Yes please! If you're still getting the `EPERM` issue on the latest release & with a cache folder outside `node_modules`, please open a new issue and tag me in it so I can have a look :) Thank you!
